stdlib/bigarray.ml

```diff
@@ -129,10 +129,64 @@ module Genarray = struct
   external blit: ('a, 'b, 'c) t -> ('a, 'b, 'c) t -> unit
      = "caml_ba_blit"
   external fill: ('a, 'b, 'c) t -> 'a -> unit = "caml_ba_fill"
+
+  external ptr : ('a, 'b, 'c) t -> nativeint = "caml_ba_ptr"
```

Comment:
This function can be `unboxed` and `noalloc` by defining a native and a bytecode version:

```ocaml
external ptr : ('a, 'b, 'c) t -> (nativeint [@unboxed]) = "caml_ba_ptr_byte" "caml_ba_ptr_opt" [@@noalloc]
```

```c
CAMLprim intnat caml_ba_ptr_opt(value va)
{
  struct caml_ba_array * a = Caml_ba_array_val(va);
  return (intnat) (a->data);
}

CAMLprim value caml_ba_ptr_byte(value va)
{
  CAMLparam1(va);
  CAMLlocal1(res);

  res = caml_copy_nativeint(caml_ba_ptr_opt(va));

  CAMLreturn(res);
}
```

This is not necessary as it's a very small part of `overlap`'s runtime but it's a nice example of unboxed stub and may be useful in the future.